modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js

```diff
@@ -69,20 +70,31 @@ const useFormSubmit = ({apiRef, containerRef}) => {
 							return;
 						}
 
-						Liferay.Util.submitForm(event.target);
+						if (submittable) {
+							Liferay.Util.submitForm(event.target);
+						}
+
+						Liferay.fire('ddmFormValid', {
+							formWrapperId: event.target.id,
+						});
 
 						Liferay.fire('ddmFormSubmit', {
 							formId: getFormId(
 								getFormNode(containerRef.current)
 							),
 						});
 					}
+					else {
+						Liferay.fire('ddmFormError', {
+							formWrapperId: event.target.id,
+						});
+					}
 				})
 				.catch((error) => {
 					console.error(error);
 				});
 		},
-		[containerRef, apiRef]
+		[containerRef, apiRef, submittable]
```

Comment:
Please, sort.

Reply:
Hey, @boton. It appears that you didn't sort this in your new push. Could you please make sure this is sorted too?